### PR TITLE
PF-1009: handle the cases when the system file path is symbolic link 

### PR DESCRIPTION
**Pull request by yuhuyoyo**
when mounting config files to k8s containers, the path is actually a symbolic link. Exception is thrown when trying to read those path.

### CODE REVIEW

File: src/test/java/bio/terra/buffer/config/PoolSchemaTest.java [hunk 1]
```diff
@@ -41,6 +42,11 @@ public void testConfigValid_readFromSystemFile() {
     }
   }
 
+  @Test
+  public void loadPoolConfig_systemFilePathIsSymbolicLink_configValid() throws IOException {
+    loadPoolConfig("config", Optional.ofNullable("./src/test/java/bio/terra/buffer/config"));
```
Comment (yonghaoy) [inline src/test/java/bio/terra/buffer/config/PoolSchemaTest.java:47]: nit: unnecessary of using `ofNullable`

**Review by jmczerk — COMMENTED** [review]
The code change looks good to me, but I'm not sure if it's OK to put config files under `src/test/java`

File: src/test/java/bio/terra/buffer/config/PoolSchemaTest.java (hunk 1 above)
Comment (yonghaoy) [inline src/test/java/bio/terra/buffer/config/PoolSchemaTest.java:47]: you can check `loadPoolConfig` result to make sure poolSchema finds all resources configs.
```
assertEqual(loadPoolConfig("config", Optional.of("./src/test/java/bio/terra/buffer/config")).size()), 2);
```

**Comment by yuhuyoyo** [comment]
> The code change looks good to me, but I'm not sure if it's OK to put config files under `src/test/java`

hmm I cannot put the symlink file under resources/ though so I just put it closest to where the test is.

**Review by yonghaoy — APPROVED** [review]

**Comment by yonghaoy** [comment]
> > The code change looks good to me, but I'm not sure if it's OK to put config files under `src/test/java`
> 
> hmm I cannot put the symlink file under resources/ though so I just put it closest to where the test is.

In Java unit test, it's common to put test data files in the same directory as the unit test to make them close to each other.

**Review by jmczerk — APPROVED** [review]

**Pull request merged**